dss/api/subscriptions.py

```diff
@@ -94,20 +90,32 @@ def put(json_request_body: dict, replica: str):
     # john@example.com would show up because elasticsearch matched example w/ example.
     # By including "index": "not_analyzed", Elasticsearch leaves all owner inputs alone.
     index_name = Config.get_es_index_name(ESIndexType.subscriptions, Replica[replica])
-    #  TODO (tsmith): get all indexes that use current alais
-    #  TODO (tsmith): try to subscribe query to each of the indexes.
-    #  TODO (tsmith): error if no queries are indexed.
     get_elasticsearch_subscription_index(es_client, index_name, logger, index_mapping)
 
-    try:
-        percolate_registration = _register_percolate(es_client, uuid, es_query, replica)
-        logger.debug(f"Percolate query registration succeeded:\n{percolate_registration}")
-    except ElasticsearchException as ex:
+    #  get all indexes that use current alias
+    alias_name = Config.get_es_alias_name(ESIndexType.docs, Replica[replica])
+    doc_indexes = _get_indexes_by_alias(es_client, alias_name)
+
+    #  try to subscribe query to each of the indexes.
+    subscribed_indexes = []
+    for doc_index in doc_indexes:
+        try:
+            percolate_registration = _register_percolate(es_client, doc_index, uuid, es_query, replica)
+        except ElasticsearchException as ex:
+            logger.debug(f"Exception occured when registering a document to an index. Exception: {ex}")
+            last_ex = ex
+        else:
+            logger.debug(f"Percolate query registration succeeded:\n{percolate_registration}")
+            subscribed_indexes.append(doc_index)
+
+    # Queries are unlikely to fit in all of the indexes, therefore errors will almost always occur. Only return an error
+    # if no queries are successfully indexed.
+    if len(doc_indexes) > 0 and not subscribed_indexes:
```

Comment:
`if doc_indexes and not subscribed_indexes` because non-empty lists are `True-ish`

I prefer this whenever it reads better to the human eye, and I think in this case it does.

Reply:
I agree